File: arena-xunit/src/ArenaHost.cs
using ArenaDotnet.Xunit.Ffi;

namespace ArenaDotnet.Xunit;

public static class ArenaHost
{
    public static int FindAvailablePort(int rangeStart, int rangeEnd, PortSearchStrategy strategy = PortSearchStrategy.Random)
    {
        return ArenaBindings.FindAvailablePort(rangeStart, rangeEnd, strategy);
    }
}

File: arena-junit/src/main/java/arena/junit/ArenaHost.java
package arena.junit;

import arena.junit.ffi.ArenaBindings;
import arena.junit.ffi.PortSearchStrategy;

public final class ArenaHost {
  private ArenaHost() {}

  public static int findAvailablePort(int rangeStart, int rangeEnd, PortSearchStrategy strategy) {
    return ArenaBindings.findAvailablePort(rangeStart, rangeEnd, strategy);
  }

  public static int findAvailablePort(int rangeStart, int rangeEnd) {
    return findAvailablePort(rangeStart, rangeEnd, PortSearchStrategy.RANDOM);
  }
}
